dsaps/dspace.py

```diff
@@ -173,96 +174,105 @@ def _build_uuid_list(self, rec_obj, children):
         return child_list
 
 
-@attr.s
-class BaseRecord:
-    uuid = Field()
-    name = Field()
-    handle = Field()
-    link = Field()
-    objtype = Field()
-
-
-@attr.s
-class Collection(BaseRecord):
-    items = Group()
-
-    def post_items(self, client):
-        """Post items to collection."""
-        for item in self.items:
-            logger.info(f"Posting item: {item}")
-            item_uuid, item_handle = client.post_item_to_collection(self.uuid, item)
-            item.uuid = item_uuid
-            item.handle = item_handle
-            logger.info(f"Item posted: {item_uuid}")
-            for bitstream in item.bitstreams:
-                bitstream_uuid = client.post_bitstream(item_uuid, bitstream)
-                bitstream.uuid = bitstream_uuid
-                logger.info(f"Bitstream posted: {bitstream_uuid}")
-            yield item
+@define
+class Bitstream:
+    name = field(default=None)
+    file_path = field(default=None)
 
-    @classmethod
-    def create_metadata_for_items_from_csv(cls, csv_reader, field_map):
-        """Create metadata for the collection's items based on a CSV and a JSON mapping
-        field map."""
-        items = [Item.metadata_from_csv_row(row, field_map) for row in csv_reader]
-        return cls(items=items)
+
+@define
+class MetadataEntry:
+    key = field(default=None)
+    value = field(default=None)
+    language = field(default=None)
 
 
-@attr.s
-class Community(BaseRecord):
-    collections = Field()
+@define
+class Object:
+    uuid = field(default=None)
+    name = field(default=None)
+    handle = field(default=None)
+    link = field(default=None)
+    type = field(default=None)
 
 
-@attr.s
-class Item(BaseRecord):
-    metadata = Group()
-    bitstreams = Group()
-    file_identifier = Field()
-    source_system_identifier = Field()
+@define
+class Item(Object):
+    metadata = field(factory=list)
+    bitstreams = field(factory=list)
+    item_identifier = field(default=None)
+    source_system_identifier = field(default=None)
 
-    def bitstreams_in_directory(self, directory, s3_client, file_type=None):
-        """Create a list of bitstreams from the specified directory and sort the list."""
-        pass
+    @classmethod
+    def create(cls, record, mapping) -> Item:
+        return cls(
+            metadata=cls.get_metadata(record, mapping),
+            bitstreams=cls.get_bitstreams(record),
+            **cls.get_ids(record, mapping),
+        )
 
     @classmethod
-    def metadata_from_csv_row(cls, row, field_map):
+    def get_bitstreams(cls, record) -> list:
+        if bitstreams := record.get("bitstreams"):
```

Comment:
Could we actually populate the `Bitstream` class here?  It feels awkward calling it in the CLI when we could just populate the class here and attach it to the `Item.bitstreams` attribute. That feels like a better use of `Item.bitstreams` than storing the list of filepaths from the CSV.

Reply:
See updates [here](https://github.com/MITLibraries/dspace-api-python-scripts/pull/39/commits/6399d40c18e371db823e7d4a80b3759fc357e3f7#diff-f5d9195cfa20a1be2e842f3ab15f1d4b7abd967bcbc745c41042b430241233e1R215-R220).